plugin/src/main/java/at/helpch/chatchat/user/UsersHolder.java

```diff
@@ -19,16 +17,17 @@ public final class UsersHolder {
 
     public static final User CONSOLE = ConsoleUser.INSTANCE;
 
-    private @NotNull final Map<UUID, User> users = new HashMap<>();
+    private final ChatChatPlugin plugin;
 
-    private @NotNull final Set<UUID> socialSpyUsers = new HashSet<>();
+    private @NotNull final Map<UUID, User> users = new HashMap<>();
 
-    public UsersHolder() {
+    public UsersHolder(@NotNull final ChatChatPlugin plugin) {
+        this.plugin = plugin;
         users.put(CONSOLE.uuid(), CONSOLE);
     }
 
     public @NotNull User getUser(@NotNull final UUID uuid) {
-        return users.computeIfAbsent(uuid, ChatUserImpl::new);
+        return users.computeIfAbsent(uuid, user -> plugin.database().loadChatUser(uuid));
```

Comment:
Not super important probably, but `user-> plugin.database().loadChatUser(uuid));`, `user` is ignored since the uuid passed to the method is used. 
Maybe rename to `__` or `ignored`. It is also a `UUID` not a `User`, so the name is a tad could be confusing.